lib/ash/resource.ex

```diff
@@ -44,22 +44,48 @@ defmodule Ash.Resource do
                         )
 
   @moduledoc """
-  The entry point for creating an `Ash.Resource`.
+  A resource is a static definition of an entity in your system.
 
-  This brings in the top level DSL macros, defines module attributes for aggregating state as
-  DSL functions are called, and defines a set of functions internal to the resource that can be
-  used to inspect them.
+  In general an entity will refer to a single data concept and use a `Data Layer`, which allow them to be persisted and manipulated via Ash.
 
-  Simply add `use Ash.Resource, ...` at the top of your resource module, and refer to the DSL
+  | Data Layer | Use Case |
+  | :--- | :--- |
+  | Postgres | Database tables ("Posts" or "Authors" or "Tweets") |
+  | Postgres | Database views (SQL statistic such as Tweets Per Hour by Country) |
+  | ETS | _________???________ |
+  | File | Imagine a large json file of thousands of cities with metadata |
+  | External | 3rd Party APIs such as Stripe |
```

Comment:
I wouldn’t necessarily list these out like this quite yet. “External” data layers would need to be implemented custom for anyone who wanted to make one. I wouldn’t expect `file` so much as `json_file` or `csv_file`

Entities that aren’t backed by a data layer wouldn’t typically be used for something like registration. Realistically, you might just use resources to *describe* something. For instance, you could use resources to auto document an API that you already made or something. But for now I’d just leave all of that out of the readme. Don’t want to put the cart too far before the horse, the existing ETS And Postgres data layers are pretty far from done as is